app/validators/blocks/block_validator.py

```diff
@@ -45,23 +46,24 @@ def validate_id_relationships_used_with_relationship_collector(self):
                 block_id=self.block["id"],
             )
 
-    def validate_source_references(self, source_references):
-        for source_reference in source_references:
-            source = source_reference["source"]
-            if isinstance(source_reference["identifier"], str):
-                identifiers = [source_reference["identifier"]]
-            else:
-                identifiers = source_reference["identifier"]
-
-            if source == "answers":
-                selector = source_reference.get("selector")
-                self.validate_answer_source_reference(identifiers, selector)
+    def validate_source_reference(self, source_reference_path, source_reference):
+        source = source_reference["source"]
+        if isinstance(source_reference["identifier"], str):
+            identifiers = [source_reference["identifier"]]
+        else:
+            identifiers = source_reference["identifier"]
+
+        if source == "answers":
+            selector = source_reference.get("selector")
+            self.validate_answer_source_reference(
+                source_reference_path, identifiers, selector
+            )
```

Comment:
I find this hard to follow with all the sources in the vars and then source_reference having a source, can we possibly make them more distinguishable? Due to the naming my mind also tells me sources_reference is an item of the list sources references, which it isn't in this instance.

Reply:
I agree, but the only thing changed here was move the loop up a level, and this is likely to be changed in the validator code changes I'm making.